bootstrap/kubeadm/api/v1beta1/kubeadm_types.go

```diff
@@ -585,7 +585,7 @@ type BootstrapTokenDiscovery struct {
 	// token is a token used to validate cluster information
 	// fetched from the control-plane.
 	// +required
-	// +kubebuilder:validation:MinLength=1
+	// +kubebuilder:validation:MinLength=0
```

Comment:
I looked into https://github.com/kubernetes-sigs/cluster-api/pull/12019 that fixed a similar issue by making the field optional and adding omit empty, which seems a cleaner option to allow users to not set a field.

@JoelSpeed WDYT?

Also, Is this a non-breaking change? (I assume it is not breaking, but I would like to double check)

Reply:
> I looked into #12019 that fixed a similar issue by making the field optional and adding omit empty, which seems a cleaner option to allow users to not set a field.
> 
Right, making it optional would be better, will wait for @JoelSpeed's response.

> Also, Is this a non-breaking change? (I assume it is not breaking, but I would like to double check)

It should be a non breaking change.

Reply:
Technically any change we would make here could be considered breaking.

Adding a min length of 0 to a required field is odd and I would avoid personally. A required field where the zero value is valid does not actually behave as a required field for those using structured clients. For those using unstructured clients, it's even weirder, they have to specify the key, and possibly then set the zero value.

The lesser of two evils here, if there needs to be a case where this field is empty, it to make the field optional with omitempty.

Who consumes this API? Do we understand what the consumers will do if they see an empty string here?

Reply:
Thank you for the response, Token is being currently used in kubeadm_config controller and we have logic written to handle both the cases, that is to respect the value if set or generate one. Reference https://github.com/kubernetes-sigs/cluster-api/blob/3e6bc6aa4903eec6f2d4e41848001aa2d5689e18/bootstrap/kubeadm/internal/controllers/kubeadmconfig_controller.go#L1270-L1284

Reply:
As per suggestions, changed to field to `optional`. PTAL.